src/Whim/Workspace/WorkspaceManager.cs

```diff
@@ -402,12 +402,35 @@ public void ActivatePrevious(IMonitor? monitor = null)
 		int idx = _workspaces.IndexOf(currentWorkspace);
 		int prevIdx = (idx - 1).Mod(_workspaces.Count);
 
+		if (skipActive)
```

Comment:
Could you separate out the functionality into separate methods? Something like `ActivateSkip` and `ActivateAdjacent` (names can be different).

I think it'd be possible to share the logic between `ActivatePrevious` and `ActivateNext` - maybe use a do-while, and provide an `int delta`.

Reply:
Sorry for the miscommunication - I liked your prior change to the interface with `void ActivateNext(IMonitor? monitor = null, bool skipActive = false);` etc.

Good point about #659. Based on that, I think a basic structure could be:

```csharp
public void ActivatePrevious(IMonitor? monitor = null, bool skipActive = false)
{
	Logger.Debug("Activating previous workspace");

	monitor ??= _context.MonitorManager.ActiveMonitor;
	IWorkspace? currentWorkspace = GetWorkspaceForMonitor(monitor);
	if (currentWorkspace == null)
	{
		Logger.Debug($"No workspace found for monitor {monitor}");
		return;
	}

	IWorkspace? prevWorkspace = GetAdjacentWorkspace(currentWorkspace, getNext: false, skipActive);
	if (prevWorkspace == null)
	{
		Logger.Debug($"No previous workspace found for monitor {monitor}");
		return;
	}

	Activate(prevWorkspace, monitor);
}

...

private IWorkspace? GetAdjacentWorkspace(IWorkspace workspace, bool getNext, bool skipActive)
{
	int idx = _workspaces.IndexOf(workspace);
	int delta = getNext ? 1 : -1;
	int nextIdx = (idx + delta).Mod(_workspaces.Count);

	while (idx != nextIdx)
	{
		IWorkspace nextWorkspace = _workspaces[nextIdx];
		IMonitor? monitor = GetMonitorForWorkspace(nextWorkspace);

		if (monitor == null || !skipActive)
		{
			return nextWorkspace;
		}

		nextIdx = (nextIdx + delta).Mod(_workspaces.Count);
	}

	return null;
}
```

DISCLAIMER: I haven't actually run this code - it's more of an idea

Reply:
I had to provide a disclaimer - otherwise I get embarrassed when the code I wrote really quickly outside an editor doesn't work 😆